data/transactions/logic/box_test.go

```diff
@@ -293,6 +368,57 @@ func TestBoxWriteBudget(t *testing.T) {
 
 }
 
+// TestWriteBudgetPut ensures we get write budget right for box_put
+func TestWriteBudgetPut(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	ep, _, ledger := logic.MakeSampleEnv()
+	ledger.NewApp(basics.Address{}, 888, basics.AppParams{})
+
+	// Sample tx[0] has two box refs, so write budget is 2*100
+
+	// Test simple use of one box, less than, equal, or over budget
+	logic.TestApp(t, `byte "self"; int 150; box_create`, ep)
+	logic.TestApp(t, `byte "self"; int 150; bzero; box_put; int 1`, ep)
+	logic.TestApp(t, `byte "self"; int 149; bzero; byte "x"; concat; box_put; int 1`, ep)
+	// puts to same name, doesn't go over budget (although we don't optimize
+	// away puts with the same content, this test uses different contents just
+	// to be sure).
+	logic.TestApp(t, `byte "self"; int 150; bzero; box_put;
+	                  byte "self"; int 149; bzero; byte "x"; concat; box_put; int 1`, ep)
+	// puts to different names do
+	logic.TestApp(t, `byte "self"; int 150; bzero; box_put;
+	                  byte "other"; int 149; bzero; byte "x"; concat; box_put; int 1`, ep,
+		"write budget")
+
+}
+
+// TestBoxRepeatedCreate ensures that app is not charged write budget for
+// creates that don't do anything.
+func TestBoxRepeatedCreate(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	ep, _, ledger := logic.MakeSampleEnv()
+	ledger.NewApp(basics.Address{}, 888, basics.AppParams{})
+
+	// Sample tx[0] has two box refs, so write budget is 2*100
+	logic.TestApp(t, `byte "self"; int 201; box_create`, ep,
+		"write budget")
+	ledger.DelBoxes(888, "self")
```

Comment:
@jannotti Minor:  Sanity checking my understanding, can the highlighted line be deleted?

Test passes locally with the change.  I think it's safe because there's no box created in the prior test.  If it's unsafe, I'd appreciate a terse explanation for my intuition.


```suggestion

```

Reply:
@jannotti Got it - I'll leave the thread open in case you prefer to take the _suggestion_.  Otherwise, feel welcomed to resolve as is.

Reply:
Remove it and its friends.